### DISCUSSION TITLE
Help needed: Python problem after libresbcinstallation

### DISCUSSION THREAD

**Discussion by poldy42** [hnimminh/libresbc#84] [Q&A] [answered]
Good evening,

looking for an alternative to a high-priced SBC for my private environment, I found libresbc very promising. After my first attempts with Debian Bullseye (Deb11) were unsuccessful, I'm currently testing around with Debian Buster (Deb10). I also managed to install via Ansible, but there seems to be a problem with newer python versions. The liberator service does not start, if I start it manually, python throws me a definition error message:
-------------------
sbcadmin@libresbc:~$ sudo /usr/bin/python3 /opt/libresbc/0.5.10-b/liberator/main.py
Traceback (most recent call last):
  File "/opt/libresbc/0.5.10-b/liberator/main.py", line 19, in <module>
    from api import httpapi
  File "/opt/libresbc/0.5.10-b/liberator/api.py", line 18, in <module>
    from configuration import _APPLICATION, _SWVERSION, _DESCRIPTION
  File "/opt/libresbc/0.5.10-b/liberator/configuration.py", line 72
    DISKCDR_ENABLE = {% if diskcdr is defined %}{{ 'True' if diskcdr else 'False'}}{% else %}False{% endif +%}
                      ^
SyntaxError: invalid syntax
-------------------
Since python and I are not really friends (yet) is every hint highly appreciated :-)

Cheers - Michael

BTW: Deb10 is on latest patch level, python version installed is 3.7

**Comment by hnimminh** [comment]
Hi @poldy42, 
It look like the issue of rendering template, can you provide the ansible install log? or you can send to me via the email if you have confidential info. thanks

**Comment by poldy42** [comment]
Hi, many thanks for your support. Just send you the logs and the complete build/ansible folder. I needed to disable copy of configuration file in task libre/liberator.yml, otherwise I go a bunch of errors, sounds like ansible tries to interpret content of configuration file instead of just copying it. 

Installation was startet with this command:
ansible-playbook playbooks/deployment.yml -i inventories/production -l "<machine-name>" -t "platform,libre,nginx,captagent" -e "signalwire_pat_token=pat_XXXXXXXXXXXXX"

If you need any further info don't hesitate to write, I a very keen to have this up and running. If you need access to the system (there is nothing secret about the system) this can be arranged via ssh.

Best regards - Michael

**Reply by hnimminh** [reply]
@poldy42 
it is not just a copy, that include render templating. could you please rerun playbook with original version and send me ansible.log.

**Reply by poldy42** [reply]
Hi,

okay, started from scratch. Plain empty Debian Buster, fresh git clone for libresbc, just host.yml and ansible.cfg modified. Please find attached ansible.log. You can find two runs in log file, first was abortet because of missing vin.basic (in my debian it's called vim.tiny).

Cheers - Michael
[ansible.log.gz](https://github.com/hnimminh/libresbc/files/10393757/ansible.log.gz)

**Reply by hnimminh** [reply]
Thanks @poldy42 May I know your ansible version ?

**Reply by poldy42** [reply]
Of course :-)
----
dpkg -l|grep ansible
ii  ansible                              2.10.7+merged+base+2.10.8+dfsg-1 all 
----
Cheers - Michael

**Reply by poldy42** [reply]
> Thanks @poldy42 May I know your ansible version ?

Hi @hnimminh , were already able to reproduce the problem somehow ?
Cheers - Michael

**Reply by hnimminh** [reply]
@poldy42  The issue has fixed for few months ago, somehow I forgot to let you know  @_@
Could please retest on latest release (or even master branch)

**Comment by anthon-walters** [comment, marked as answer]
For someone who may stumble onto this thread. I had a similar issue after an ansible install on Debian 11 (bullseye). It turned out that the /usr/bin/python was symlinked to Python2.7 instead of /usr/bin/python3.9.
